src/napari_threedee/annotators/dipoles/annotator.py

```diff
@@ -155,6 +155,10 @@ def _set_direction_from_mouse_event(self, event: Event = None):
         )
         dipole_position_3d = self.active_dipole_position[displayed_dims]
         direction = dipole_north_position_3d - dipole_position_3d
+        norm = np.linalg.norm(direction)
+        if norm == 0:
```

Comment:
Working with floating point numbers is funny, see here for details https://bitbashing.io/comparing-floats.html

Let's instead check that the absolute difference between norm and 0 is less than some threshold epsilon 🙂